### ISSUE TITLE
Multi-GPU not working when using model(inputs) nor when computing the gradients

### ISSUE THREAD

**Issue by miguelCalado** [keisen/tf-keras-vis#70]
Hi, 

I've been using ``tf-keras-vis`` for a while and wanted to speed things up by using more than one GPU to compute the saliency maps of multiple images. I noticed that when using the interpretability methods (the ones that use ``tape.gradients`` and ``model(inputs)``) only one GPU is being used.

I don't know if this is normal behavior, a bug on TensorFlow's part, or a problem on my end, but I decided to open an issue, maybe someone has tried it. 

Here are a couple of useful links:

- [Stackoverflow](https://stackoverflow.com/questions/68283519/tensorflow-multi-gpu-doesn-t-work-for-modelinputs-nor-when-computing-the-gra): I opened this issue on StackOverflow and added a bounty to it. You can find a list of things that I've tried and also more information on this issue.
- [Working Example](https://drive.google.com/file/d/1CoYVckmEQXp2Wf_PRlGtnrlGnF8smxvt/view?usp=sharing): A trained CNN on the ``dogs_vs_cats`` dataset. This replicates the problem of not being able to distribute the performance between the GPUs. You can also download the models in [HDF5](https://drive.google.com/file/d/1Y0-fQytVsnHPs8JL6kKJr3tEL0mKNtjJ/view?usp=sharing) and [Saved](https://drive.google.com/file/d/19oSIaUTtEy1q6rlDj8GzuWIvwAq_7XMi/view?usp=sharing) formats.

**Comment by keisen** [comment]
@miguelCalado , Thank you for letting us know that and sharing links! They will be great helps for us.

Unfortunately, `tf-keras-vis` has never been tested on a machine that has multiple GPUs.
I don't know the cause of the problem you're facing.
If I get an environment where can be used multiple GPUs, I can tackle this problem.
However, it will take some time to do so.

Thanks!
